core/src/main/kotlin/org/lflang/generator/cpp/CppGenerator.kt

```diff
@@ -94,14 +94,48 @@ class CppGenerator(
             context.reportProgress(
                 "Code generation complete. Compiling...", IntegratedBuilder.GENERATED_PERCENT_PROGRESS
             )
-            if (platformGenerator.doCompile(context)) {
-                context.finish(GeneratorResult.Status.COMPILED, codeMaps)
+            if (targetConfig.get(DockerProperty.INSTANCE).enabled) {
+                copySrcGenBaseDirIntoDockerDir()
+                buildUsingDocker()
             } else {
-                context.unsuccessfulFinish()
+                if (platformGenerator.doCompile(context)) {
+                    context.finish(GeneratorResult.Status.COMPILED, codeMaps)
+                } else {
+                    context.unsuccessfulFinish()
+                }
             }
         }
     }
 
+    /**
+     * Copy the contents of the entire src-gen directory to a nested src-gen directory next to the generated Dockerfile.
+     */
+    private fun copySrcGenBaseDirIntoDockerDir() {
```

Comment:
This is a very hacky solution. I wish there would be an easy way to avoid the copying. The fact that the copying is needed is symptomatic of a more profound issue in the way the normal code generator and the docker generator interact. If the docker generator needs sources to be in a specific location, it should pass an according file config to the C++ generator, and it will do that. However, this doesn't work in the current architecture, as everything goes through GeneratorBase and the target code generator ends up calling the docker generator, and not the other way around.

Reply:
I agree with you...